cli/src/test/scala/com/codacy/analysis/cli/formatter/SarifSpec.scala

```diff
@@ -0,0 +1,98 @@
+package com.codacy.analysis.cli.formatter
+
+import java.io.{ByteArrayOutputStream, PrintStream}
+
+import better.files.File
+import com.codacy.analysis.core.model.{Issue, LineLocation}
+import com.codacy.plugins.api.PatternDescription
+import com.codacy.plugins.api.results.{Pattern, Result, Tool}
+import io.circe.generic.auto._
+import io.circe.syntax._
+import io.circe.{Encoder, Printer}
+import org.specs2.control.NoLanguageFeatures
+import org.specs2.mutable.Specification
+
+class SarifSpec extends Specification with NoLanguageFeatures {
+
+  private implicit val levelEncoder: Encoder[SarifReport.Level.Value] = Encoder.encodeEnumeration(SarifReport.Level)
+
+  "Sarif" should {
+
+    "output correct json in sarif format" in {
+      File.temporaryDirectory("sarif-formatter") { sourceDirectory =>
+        // Arrange
+        val sourceFile = sourceDirectory / "test.js" write "line 1\nline 2"
+        val patternId = Pattern.Id("custom-pattern-id")
+        val patternLevel = Result.Level.Err
+        val patternCategory = Pattern.Category.Security
+        val toolSpecification = Tool.Specification(
+          Tool.Name("Custom Tool"),
+          Option(Tool.Version("1.0.0")),
+          Set(Pattern.Specification(patternId, patternLevel, patternCategory, None)))
+        val patternDescriptions = PatternDescription(
+          patternId,
+          "My Pattern Title",
+          None,
+          Option("My Pattern Description"),
+          None,
+          Option("My Pattern Explanation"))
+        val issue = Issue(
+          patternId,
+          sourceFile.path.getFileName,
+          Issue.Message("My Issue Message"),
+          patternLevel,
+          Option(patternCategory),
+          LineLocation(1))
+
+        val expectedArtifactLocation = s"file:///codacy/${issue.filename.toString}"
+        val expectedReport = SarifReport(
```

Comment:
Can you move this to a private method to improve the readability of this test?

Reply:
I can, but since this is an important part of the expectations of the test and I do not plan to repeat it, I think having it colocated makes it more readable.